src/Manager.cpp

```diff
@@ -447,4 +447,17 @@ Manager::sequence(uint32_t queueIndex, uint32_t totalTimestamps)
     return sq;
 }
 
+DeviceProperties Manager::getDeviceProperties() const
+{
+  const vk::PhysicalDeviceProperties properties = this->mPhysicalDevice->getProperties();
+  const DeviceProperties output{
```

Comment:
The idea was that most of the properties are not relevant for Kompute. Moreover, some of the properties could be post-processed for easier access by the user, e.g. amount of device memory: Using the raw `vk::` struct one would need to find the correct memory type first.

Reply:
Not sure I understand, what would be the requirements to find the correct memory type? I can see that you read for example `properties.limits.maxComputeWorkGroupCount`, is the memory type whether it's not of type Compute? I agree that there is a large number of options on the device properties, I'm just thinking that having a struct in between may result in more abstractions to maintain and potentially people requesting more to be exposed - what about the suggestion on having it as a python dict? It could be buitl as a single-level python dict, where the key could just be limits_max_compute_work_group_count, etc

Reply:
This is not yet implemented. I mean that the the user would need to iterate over the memory types as in `Tensor::allocateBindMemory` to find the correct memory heap that is used for the buffers.